src/nvim/option.c

```diff
@@ -215,6 +216,29 @@ void set_init_1(bool clean_arg)
     }
   }
 
+  /*
+   * Find default value for 'spelllang' option.
+   * Don't use it if it is empty.
+   */
+  {
+    const char *lang = os_getenv("LANG");
```

Comment:
there is already code checking for current locale. it's tricky.

and this suggests what I suspected, which is that this is not worth the effort.

there is a cost also: documentation and troubleshooting become less predictable.

Reply:
The 'shell' option is set basically the same way. How does it make sense to set the 'shell' option to the user's default shell but if it does not make sense to set the 'spelllang' option the user's default language?